### PR TITLE
Add yamllint in kapitan linter checks

### PR DESCRIPTION
**Pull request by uberspot**
Adds a yaml lint check for every file in a given inventory to help debug yaml issues.
Uses a very relaxed yamllint config by default but can be overriden with the rules in a local '.yamlllint' file if that file exists.

Rules taken from https://yamllint.readthedocs.io/en/stable/rules.html

### CODE REVIEW

File: kapitan/cli.py [hunk 1]
```diff
@@ -24,24 +24,22 @@
...
-import yaml
 
-from kapitan.utils import jsonnet_file, PrettyDumper, flatten_dict, searchvar
-from kapitan.utils import deep_get, from_dot_kapitan, check_version, fatal_error
-from kapitan.utils import search_target_token_paths
-from kapitan.targets import compile_targets
-from kapitan.resources import search_imports, resource_callbacks, inventory_reclass
-from kapitan.version import PROJECT_NAME, DESCRIPTION, VERSION
-from kapitan.lint import start_lint
+import yaml
```
Comment (uberspot) [inline kapitan/cli.py]: These were automatically reordered by my IDE.

File: kapitan/lint.py [hunk 2]
```diff
@@ -148,3 +187,40 @@ def lint_unused_classes(inventory_path):
...
+
+                try:
+                    problems = list(linter.run(file_contents, conf))
+                except EnvironmentError as e:
+                    logger.error(e)
+                    sys.exit(-1)
+  
+                if len(problems) > 0:
+                    checks_sum += len(problems)
+                    logger.info("File {} has the following issues:\n".format(path))
```
Comment (gburiola) [inline kapitan/lint.py]: output looks better if you remove the `\n`

BEFORE

```
Running yamllint on all inventory files...

File ./inventory/targets/file1.yml has the following issues:

    47:9: duplication of key "key1" in mapping (key-duplicates)
File ./inventory/targets/file2.yml has the following issues:

    63:5: duplication of key "key2" in mapping (key-duplicates)

Total yamllint issues found: 2

Checking for orphan classes in inventory...
```

AFTER

```
Running yamllint on all inventory files...

File ./inventory/targets/file1.yml has the following issues:
	47:9: duplication of key "key1" in mapping (key-duplicates)
File ./inventory/targets/file2.yml has the following issues:
	63:5: duplication of key "key2" in mapping (key-duplicates)

Total yamllint issues found: 2

Checking for orphan classes in inventory...
```

File: kapitan/lint.py [hunk 3]
```diff
@@ -148,3 +187,40 @@ def lint_unused_classes(inventory_path):
...
+    checks_sum = 0
+    for path in list_all_paths(inventory_path):
+        if os.path.isfile(path) and (path.endswith(".yml") or path.endswith(".yaml")):
+            with open(path, "r") as yaml_file:
+                file_contents = yaml_file.read()
+
+                if os.path.isfile('.yamllint'):
+                    conf = YamlLintConfig(file='.yamllint')
+                else:
+                    conf = YamlLintConfig(yamllint_config)
```
Comment (gburiola) [inline kapitan/lint.py]: we can also print `logger.info(".yamllint not found. Using default values")

File: kapitan/lint.py [hunk 4]
```diff
@@ -148,3 +183,42 @@ def lint_unused_classes(inventory_path):
...
+        conf = YamlLintConfig(yamllint_config)
+
+    checks_sum = 0
+    for path in list_all_paths(inventory_path):
+        if os.path.isfile(path) and (path.endswith(".yml") or path.endswith(".yaml")):
+            with open(path, "r") as yaml_file:
+                file_contents = yaml_file.read()
+
+                try:
+                    problems = list(linter.run(file_contents, conf))
```
Comment (gburiola) [inline kapitan/lint.py]: add `filepath` to `list(linter.run(file_contents, conf, filepath=path))`
https://www.pydoc.io/pypi/yamllint-1.8.1/autoapi/linter/index.html#linter.linter.run

This way we can whitelist specific files from specific rules
https://yamllint.readthedocs.io/en/stable/configuration.html#ignoring-paths

Comment (uberspot) [reply kapitan/lint.py]: Done

**Pull request merged**